src/plugins/vis_type_vislib/public/vislib/lib/vis_config.js

```diff
@@ -41,7 +41,7 @@ export class VisConfig {
 
     const visType = visTypes[visConfigArgs.type];
     const typeDefaults = visType(visConfigArgs, this.data);
-    this._values = _.defaultsDeep({}, typeDefaults, DEFAULT_VIS_CONFIG);
+    this._values = _.defaultsDeep({ ...typeDefaults }, DEFAULT_VIS_CONFIG);
```

Comment:
I thought we were going to add a special case to keep `style` parameters?

Reply:
Hey @majagrubic, it made sense when we wanted to change it for 
```
this._values = {
... DEFAULT_VIS_CONFIG
...typeDefaults,
style:{
... DEFAULT_VIS_CONFIG.style
...typeDefaults.style,
}
}
```
but as I found a way to use `defaultsDeep` without perf degradation (not using 3 params but just two) the lodash function will do the above for free :)